test/python/picongpu/quick/picmi/layout.py

```
@@ -14,20 +14,36 @@ class TestPicmiPseudoRandomLayout(unittest.TestCase):
     def test_basic(self):
         """simple translation"""
         layout = picmi.PseudoRandomLayout(n_macroparticles_per_cell=7)
+        layout.ppc = 1
         layout.check()
 
-    def test_not_translated(self):
-        """pseudo random layout is not translated directly itself"""
-        layout = picmi.PseudoRandomLayout(n_macroparticles_per_cell=7)
-        with self.assertRaises(AttributeError):
-            layout.get_as_pypicongpu()
-
     def test_invalid(self):
         """erros for invalid params entered"""
         with self.assertRaisesRegex(Exception, ".*per.*"):
             layout = picmi.PseudoRandomLayout(n_macroparticles=700)
+            layout.ppc = 1
```

why do you need to set ppc to 1 here? What should throw the Exception, the constructor or the check method?

---

~~Indeed this might not be needed. It's the wrong argument `n_macroparticles` as opposed to `n_macroparticles_per_cell` or something like that, that's triggering here (and already in the first line). I might have felt to put it there as to make sure that we're not triggering the "wrong" exception in a sense (although that can't happen here) but in either case, I certainly didn't spend much time thinking about this test.~~

Sorry, just realised that it's the `check` on the line below that's actually triggering. In this case, the `ppc=1` is needed to have a unique reason for failure.

---

oh ok so it only checks the kwarg value not the member?